src/Core.tsx

```diff
@@ -341,16 +351,18 @@ export class Core extends React.Component<IModalProps, IModalState> {
       // reset state
       this.setState(INITIAL_STATE)
     }
-
+    // this fetches all available languages in this form [{en:english},...]
+    const availableLanguages = Object.entries(i18next.services.resourceStore.data).map(keyValueLanguage => { return { code: keyValueLanguage[0], name: keyValueLanguage[1].name.toString() } })
```

Comment:
We don't need to calculate this in each `render()`. We can do it only once outside the method scope.